### PR TITLE
Nightlies Dhall files

### PR DESCRIPTION
**Pull request by AryehAlexander**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Explain your changes here.

Explain how you tested your changes here.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

**Review by 0x0I — APPROVED** [review]

**Review by bkase — APPROVED** [review]

File: buildkite/scripts/deploy-nightly.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+echo "Nightly ENV: $NIGHTLY"
+echo "DAEMON_TAG: $DAEMON_TAG"
+echo "ARCHIVE_TAG: $ARCHIVE_TAG"
+
+if [ ! -z $NIGHTLY ]; then
+  echo "Deploying Nightly"
+
+  # Authenticate terraform for gcp
+  echo "$BUILDKITE_GS_APPLICATION_CREDENTIALS_JSON" > /tmp/gcp_creds.json
```
Comment (0x0I) [inline buildkite/scripts/deploy-nightly.sh:21]: Hey these can be removed if running from an agent. Probably best to just at most assert the envvars are set properly.

File: buildkite/src/Jobs/Release/MinaArtifact.dhall [hunk 2]
```diff
@@ -33,6 +33,7 @@ Pipeline.build
           S.exactly "buildkite/scripts/build-artifact" "sh",
           S.exactly "buildkite/scripts/connect-to-testnet-on-develop" "sh",
           S.exactly "dockerfiles/scripts/healthcheck-utilities" "sh",
+          S.exactly "buildkite/nightly" "txt",
```
Comment (0x0I) [inline buildkite/src/Jobs/Release/MinaArtifact.dhall:36]: This works for sure though feels like the added dummy "nightly.txt" file could be replaced with the Dhall file itself.

File: buildkite/src/Jobs/Test/Nightly.dhall [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    spec =
+      JobSpec::{
+      dirtyWhen = [
+        S.exactly "buildkite/nightly" "txt"
+      ],
+      path = "Test",
+      name = "Nightly"
+    },
+    steps = [
+      DeployNightly.step dependsOn
```
Comment (0x0I) [inline buildkite/src/Jobs/Test/Nightly.dhall:26]: Thinking we can consolidate Commands and make use of DeployTestnet here (with input parameters specific to the *Nightly* setup) to remove some duplication.

File: buildkite/src/Jobs/Release/ArchiveNodeArtifact.dhall [hunk 4]
```diff
@@ -27,6 +27,8 @@ Pipeline.build
     spec =
       JobSpec::{
         dirtyWhen = [
+          S.strictlyStart (S.contains "buildkite/src/Jobs/Release/ArchiveNodeArtifact"),
+          S.exactly "buildkite/nightly" "txt",
```
Comment (0x0I) [inline buildkite/src/Jobs/Release/ArchiveNodeArtifact.dhall:31]: Yea seems clearer to trigger on a Job versus some txt file tho up in the air for me.

File: automation/terraform/testnets/nightly/main.tf [hunk 5]
```diff
@@ -69,7 +69,7 @@ module "ci_testnet" {
   providers = { google = google.google-us-east4 }
   source    = "../../modules/kubernetes/testnet"
 
-  k8s_context = "gke_o1labs-192920_us-east4_coda-infra-east4"
+  k8s_context = "gke_o1labs-192920_us-east1_buildkite-infra-east1"
```
Comment (0x0I) [inline automation/terraform/testnets/nightly/main.tf:72]: probably worth making the k8s ctx variable

File: automation/terraform/testnets/nightly/main.tf [hunk 6]
```diff
@@ -69,7 +69,7 @@ module "ci_testnet" {
   providers = { google = google.google-us-east4 }
```
Comment (0x0I) [inline automation/terraform/testnets/nightly/main.tf:69]: this can be removed

File: buildkite/src/Command/DeployNightly.dhall [hunk 7]
```diff
@@ -0,0 +1,31 @@
+let Prelude = ../External/Prelude.dhall
+
+let Command = ./Base.dhall
+let Size = ./Size.dhall
+
+let Cmd = ../Lib/Cmds.dhall
+-- do we need this?
```
Comment (0x0I) [inline buildkite/src/Command/DeployNightly.dhall:7]: The dependency? Only if we don't have an exact image to use. Based on the current tagging system, tough to make use of a static entry that is reliably up-to-date. We could add a latest or nightly etc tag to the builds tho to get around this.

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Comment by nholland94** [comment]
Out of date

**Pull request closed without merging**